modules/apps/dataset/source-formatter-suppressions.xml

```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0"?>
+
+<suppressions>
+	<source-check>
+		<suppress checks="JavaAPISignatureCheck" files="dataset-api/src/main/java/com/liferay/dataset/data/DatasetDataJSONFactory\.java" />
```

Comment:
What's the problem with these suppressions?

Reply:
The `HttpServletRequest` parameters. It's explained in the commit message.

Reply:
Yes, I just don't understand it 😂

Reply:
I mean it, it's a real question... what is it checking and what does it expect? 🤔

Reply:
Ah, OK. Let me paste you the link referring to the SF error...

Reply:
It's explained here :point_right: https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/checks/java_api_signature_check.markdown

Reply:
You cannot use HttpServletRequest in `-api` modules unless they have a special naming. From my point of view the rule is a bit too strict. I mean, it makes sense to pass the real things instead of the HttpServletRequest when the API wants to obtain something from it. But there are many cases (like these ones) where you would need the whole `HttpServletRequest`.

These interfaces are implemented by consumers and we pass them the whole request because they are inside a request cycle and we don't know what they need.

However, we may try to think a different way of doing it if you think it makes sense...

Reply:
> the following exceptions are allowed:
>
> The name of the class contains any of the following:
>
> `Action`, `Checker`, `Cookie` etc

Well, the obvious answer is to rename this to `DatasetDataJSONCookieFactory`.

Reply:
:clap:

Reply:
`Cookie Factory`

![smell](https://media.giphy.com/media/21S5JsbRPSzN9WD8Ho/giphy.gif)